core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -444,9 +444,12 @@ protected DeterministicKeyChain(KeyCrypter crypter, KeyParameter aesKey, Determi
         basicKeyChain.importKey(rootKey);
 
         for (int i = 1; i < getAccountPath().size(); i++) {
-            encryptNonLeaf(aesKey, chain, rootKey, getAccountPath().subList(0, i));
+            DeterministicKey parent = hierarchy.get(getAccountPath().subList(0, i - 1), false, false);
+            encryptNonLeaf(aesKey, chain, parent, getAccountPath().subList(0, i));
```

Comment:
```java
encryptNonLeaf(aesKey, chain, rootKey, getAccountPath().subList(0, i));
```
In the original code, the `rootKey` (`M/`) was being used as the parent for all of the children in the account path.

So for `M/44H/5H/0H`, `M` was used as the parent key for 44H, 5H and 0H.  However, the parent of 0H is 5H, not M.  

One could think of this as a tree where each nodes parent was set to root, instead of the actual parent node.

This was an oversight in `bitcoinj`

It turns out that if the account was a BIP32 account, then its path is `M/0H` -- with such a short path, the parents are set properly.  

For any account that has a path size of greater than 2, there will be an issue.